src/cdk/drag-drop/drag-drop-registry.ts

```diff
@@ -115,18 +116,27 @@ export class DragDropRegistry<I, C extends {id: string}> implements OnDestroy {
       const isTouchEvent = event.type.startsWith('touch');
       const moveEvent = isTouchEvent ? 'touchmove' : 'mousemove';
       const upEvent = isTouchEvent ? 'touchend' : 'mouseup';
+      const upConfig = {handler: e => this.pointerUp.next(e)};
 
       // We explicitly bind __active__ listeners here, because newer browsers will default to
       // passive ones for `mousemove` and `touchmove`. The events need to be active, because we
       // use `preventDefault` to prevent the page from scrolling while the user is dragging.
       this._globalListeners
         .set(moveEvent, {handler: e => this.pointerMove.next(e), options: activeEventOptions})
-        .set(upEvent, {handler: e => this.pointerUp.next(e)})
-        .forEach((config, name) => {
-          this._ngZone.runOutsideAngular(() => {
-            this._document.addEventListener(name, config.handler, config.options);
-          });
+        .set(upEvent, upConfig);
+
+      if (isTouchEvent) {
+        // Treat `touchcancel` events the same as `touchend`. `touchcancel` will fire for cases
+        // like an OS-level event interrupting the touch sequence or the user putting too many
+        // finger on the screen at the same time.
+        this._globalListeners.set('touchcancel', upConfig);
```

Comment:
It does, but I think that this is what we want since it's the event that fires instead of `touchend` if something like an OS popup interrupts the dragging.

Reply:
Would it be difficult to abort the drop for this case? I would think that's the safest course of action, since we'll never know whether the drop will be committing some meaningful action (like sending a missile alert for Hawaii...)

Reply:
Currently we have most the necessary info, but we don't have an API to abort a drag. I could set something up, but it means that there would be another event that the consumer needs to be aware about and that even can only realistically fire on a touch device.

Reply:
I think it would still be better to abort though; even though it's an edge case, I except any "cancelled" action to have no effect